http/src/main/scala/org/http4s/blaze/http/http2/server/AbstractBodyWriter.scala

```diff
@@ -25,6 +23,8 @@ private abstract class AbstractBodyWriter(private var hs: Headers) extends BodyW
 
   protected def flushMessage(msg: Seq[StreamFrame]): Future[Unit]
 
+  protected def fail(cause: Throwable): Unit
```

Comment:
Here the same as above: can we restrict this possibly so we don't go into throwable territory?